cmd/broadcast/main.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"flag"
+	"github.com/harmony-one/libunison/internal/ida/coopcast"
+	"github.com/harmony-one/libunison/internal/ida/manycast"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net"
+	"strconv"
+	"time"
+)
+
+// InitCoopCastNode creates a coopcast node
+func InitCoopCastNode(confignbr string, configallpeer string, t0 float64, t1 float64, t2 float64, base float64, hop int) *coopcast.Node {
+	rand.Seed(time.Now().UTC().UnixNano())
+	config1 := NewConfig()
+	err := config1.ReadConfigFile(confignbr)
+	if err != nil {
+		log.Printf("unable to read config file %v", confignbr)
+	}
+	selfPeer, peerList, _ := config1.GetPeerInfo()
+	config2 := NewConfig()
+	err = config2.ReadConfigFile(configallpeer)
+	if err != nil {
+		log.Printf("unable to read config file %v", configallpeer)
+	}
+
+	_, _, allPeers := config2.GetPeerInfo()
+	Cache := make(map[coopcast.HashKey]*coopcast.RaptorQImpl)
+	SenderCache := make(map[coopcast.HashKey]bool)
+	PeerDecodedCounter := make(map[coopcast.HashKey]map[int]int)
+	node := coopcast.Node{SelfPeer: selfPeer, PeerList: peerList, AllPeers: allPeers, Cache: Cache, PeerDecodedCounter: PeerDecodedCounter, SenderCache: SenderCache, InitialDelayTime: t0, MaxDelayTime: t1, ExpBase: base, RelayTime: t2, Hop: hop}
+	return &node
+}
+
+// InitManyCastNode creates a manycast node
+func InitManyCastNode(confignbr string, configallpeer string) *manycast.Node {
+	config1 := NewConfig()
+	err := config1.ReadConfigFile(confignbr)
+	if err != nil {
+		log.Printf("unable to read config file %v", confignbr)
+	}
+	selfPeer, peerList, _ := config1.GetPeerInfo()
+	config2 := NewConfig()
+	err = config2.ReadConfigFile(configallpeer)
+	if err != nil {
+		log.Printf("unable to read config file %v", configallpeer)
+	}
+	_, _, allPeers := config2.GetPeerInfo()
+	node := manycast.Node{SelfPeer: selfPeer, PeerList: peerList, AllPeers: allPeers}
+	return &node
+}
+
+func main() {
+	graphConfigFile := flag.String("graph_config", "graph0.txt", "file containing network structure")
+	generateConfigFiles := flag.Bool("gen_config", false, "whether to generate config files from graph_config file")
+	broadCast := flag.Bool("broadcast", false, "whether to broadcast a message")
+	msgFile := flag.String("msg_file", "test.txt", "message file to broadcast")
+	configFile := flag.String("nbr_config", "configs/config_0.txt", "config file contains neighbor peers")
+	allPeerFile := flag.String("all_config", "configs/config_allpeers.txt", "config file contains all peer nodes info")
+	mode := flag.String("mode", "coopcast", "choose broadcast testing mode, [coopcast|manycast]")
+	t0 := flag.String("t0", "5", "initial delay time for symbol broadcasting")
```

Comment:
Consider using `flag.Float64`, `flag.Int`, etc., instead of using `flag.String` and later hand-parsing them using `strconv.*`.

Reply:
good point!